Libs/sith/Engine/sithPhysics.c

```diff
@@ -142,24 +148,240 @@ void J3DAPI sithPhysics_FindWaterSurface(SithThing* pThing)
+                    return;
+                }
+
+                if ((attachflags & SITH_ATTACH_CLIMBSURFACE) != 0)
+                {
+                    sithPhysics_UpdateClimbingThingPhysics(pThing, secDeltaTime);
+                    return;
+                }
+
+                sectorflags = pThing->pInSector->flags;
+                if ((sectorflags & SITH_SECTOR_UNDERWATER) != 0 || (sectorflags & SITH_SECTOR_AETHERIUM) != 0)// 0x100 - SITH_SECTOR_AETHERIUM
+                {
+                    if (pThing->type == SITH_THING_PLAYER || (sectorflags & 0x100) == 0)// 0x100
+                    {
+                        sithPhysics_UpdateUnderwaterThingPhysics(pThing, secDeltaTime);
+                        return;
+                    }
+                }
+
+                else if (pThing->type == SITH_THING_PLAYER)
+                {
+                    sithPhysics_UpdatePlayerPhysics(pThing, secDeltaTime);
+                    return;
+                }
+
+                sithPhysics_UpdateThingPhysics(pThing, secDeltaTime);
+                return;
+            }
+
+            sithPhysics_UpdateMineCarPhysics(pThing, secDeltaTime);
+
+            if ((((uint8_t)sithMain_g_frameNumber + (pThing->idx & 0xFF)) & 7) == 0)
+            {
+                sithAIAwareness_CreateTransmittingEvent(pThing->pInSector, &pThing->pos, 2, 2.0f, pThing);
+            }
+        }
+    }
 }
 
 void J3DAPI sithPhysics_ApplyForce(SithThing* pThing, const rdVector3* force)
 {
-    J3D_TRAMPOLINE_CALL(sithPhysics_ApplyForce, pThing, force);
+    SITH_ASSERTREL(pThing != ((void*)0));
+    SITH_ASSERTREL(pThing->moveType == SITH_MT_PHYSICS);
+
+    if (pThing->moveInfo.physics.mass > 0.0 && (force->x != 0.0 || force->y != 0.0 || force->z != 0.0)) {
+
+        double invMass = 1.0 / pThing->moveInfo.physics.mass;
+
+        double velY = 0.0;
+        double velZ = 0.0;
+
+        SithPhysicsFlags physflags;
+        float xForce = force->x * invMass;
+        float yForce = force->y * invMass;
+        float zForce = force->z * invMass;
+
+        velY = yForce + pThing->moveInfo.physics.velocity.y;
+        velZ = zForce + pThing->moveInfo.physics.velocity.z;
+        pThing->moveInfo.physics.velocity.x = xForce + pThing->moveInfo.physics.velocity.x;
+
+        physflags = pThing->moveInfo.physics.flags;
+
+        pThing->moveInfo.physics.velocity.y = velY;
+
+        physflags |= SITH_PF_UNKNOWN_8000;
+
+        pThing->moveInfo.physics.velocity.z = velZ;
+
+        pThing->moveInfo.physics.flags = physflags;
+    }
 }
 
 void J3DAPI sithPhysics_SetThingLook(SithThing* pThing, const rdVector3* pNormal, float secDeltaTime)
 {
-    J3D_TRAMPOLINE_CALL(sithPhysics_SetThingLook, pThing, pNormal, secDeltaTime);
+    double lookAngle = 0;
+    double absLookAngle = 0;
+
+    SITH_ASSERTREL(pThing != ((void*)0));
+	// 90� -> 1, 0� -> 0
+    lookAngle = 1.0f - (rdVector_Dot3(&pNormal, &pThing->orient.uvec));
```

Comment:
The `pNormal` is already pointer type here and getting its address will result in invalid indirection - `rdVector3**`.
Mostlikely this is also `srcmaid` fault as it doesn't recognize variable types.

Should be: `lookAngle = 1.0f - (rdVector_Dot3(pNormal, &pThing->orient.uvec));`